farm/modeling/predictions.py

```diff
@@ -1,77 +1,166 @@
 from farm.utils import span_to_string
+from abc import ABC
+from typing import List, Optional
+from pydantic import BaseModel
 
-class Span:
+class Pred(ABC):
     def __init__(self,
-                 start,
-                 end,
-                 score=None,
+                 id,
+                 prediction,
+                 context):
+        self.id = id
+        self.prediction = prediction
+        self.context = context
+
+    def to_json(self):
+        raise NotImplementedError
+
+class QAAnswer(BaseModel):
```

Comment:
1. Suggestion: should we rename this to QACandidate, to better differentiate it from QAPred?

2. Why doesnt this class have an init method and no attributes?

Reply:
QACandidate would be better. No init methods is because it is a pydantic.BaseModel object. That's just the syntax they use